precompiles/staking/staking_test.go

```diff
@@ -394,6 +401,65 @@ func allowStaking(t *testing.T, ts testSuite, amount *big.Int) {
 	require.True(t, allowed)
 }
 
+func stakeThroughCosmosAPI(
+	t *testing.T,
+	ctx sdk.Context,
+	bankKeeper bankkeeper.Keeper,
+	stakingKeeper stakingkeeper.Keeper,
+	validator stakingtypes.Validator,
+	staker sdk.AccAddress,
+	amount math.Int,
+) {
+	// Coins to stake with default cosmos denom.
+	coins := sdk.NewCoins(sdk.NewCoin("stake", amount))
```

Comment:
Not required for the pr, but might be good to explore the option of setting the stake denom to azeta .
I think it can be set using the staking params